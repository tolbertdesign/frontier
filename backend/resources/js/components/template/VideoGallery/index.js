import VideoGallery from './VideoGallery.vue'

export default VideoGallery
