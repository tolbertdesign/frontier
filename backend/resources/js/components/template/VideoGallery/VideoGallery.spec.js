import { shallowMount, createLocalVue } from '@vue/test-utils'
import VideoGallery from './VideoGallery.vue'
import { FontAwesomeIcon } from '@fortawesome/vue-fontawesome'

const localVue = createLocalVue()

const video1 = {
  external_video_id: '12345678',
  source: 'vimeo',
}

const video2 = {
  external_video_id: '98765432',
  source: 'youtube',
}

const videos = [
  video1,
  video2,
]

describe('VideoGallery', () => {
  it('Snapshop test for VideoGallery', () => {
    const wrapper = shallowMount(VideoGallery, {
      localVue,
      components: {
        FontAwesomeIcon,
      },
      propsData: {
        videos,
        selectedVideoId: video2.external_video_id,
      },
    })
    expect(wrapper).toMatchSnapshot()
  })
})
