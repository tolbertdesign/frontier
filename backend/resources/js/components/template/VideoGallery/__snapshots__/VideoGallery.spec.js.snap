// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoGallery Snapshop test for VideoGallery 1`] = `
<div class="lg:flex lg:flex-wrap -mx-2">
  <div class="p-2 lg:w-1/2">
    <article class="media border border-light hover:border-secondary cursor-pointer"><button>
        <figure class="relative media-left mr-0">
          <figure class="image is-16by9 w-32 sm:w-48" style="background-image: url(undefined);"></figure>
          <div class="absolute m-auto text-center w-1/2" style="top: 50%; left: 50%; transform: translate(-50%,-50%);">
            <fontawesomeicon-stub icon="play-circle" size="2x"></fontawesomeicon-stub>
          </div>
        </figure>
      </button>
      <div class="media-content p-2 md:p-4">
        <h3 class="text-sm sm:text-base font-semibold">

        </h3>
      </div>
    </article>
  </div>
  <div class="p-2 lg:w-1/2">
    <article class="media border border-light hover:border-secondary cursor-pointer border-secondary border-2"><button>
        <figure class="relative media-left mr-0">
          <figure class="image is-16by9 w-32 sm:w-48" style="background-image: url(undefined);"></figure>
          <div class="absolute m-auto text-center w-1/2" style="top: 50%; left: 50%; transform: translate(-50%,-50%);">
            <fontawesomeicon-stub icon="play-circle" size="2x"></fontawesomeicon-stub>
          </div>
        </figure>
      </button>
      <div class="media-content p-2 md:p-4">
        <h3 class="text-sm sm:text-base font-semibold">

        </h3>
      </div>
    </article>
  </div>
</div>
`;
